addons/pos_self_order/static/src/ProductList/ProductList.js

```diff
@@ -0,0 +1,217 @@
+/** @odoo-module */
+
+import { Component, onMounted, useEffect, useRef, useState } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/SelfOrderService";
+import { useAutofocus } from "@web/core/utils/hooks";
+import { formatMonetary } from "@web/views/fields/formatters";
+import { NavBar } from "@pos_self_order/NavBar/NavBar";
+import { fuzzyLookup } from "@web/core/utils/search";
+export class ProductList extends Component {
+    static template = "ProductList";
+    static components = { NavBar };
+    setup() {
+        this.state = useState(this.env.state);
+        this.privateState = useState({
+            selectedTag: "",
+            searchIsFocused: false,
+            searchInput: "",
+            navbarIsShown: true,
+            scrollingDown: false,
+            scrolling: false,
+        });
+        this.selfOrder = useSelfOrder();
+        this.formatMonetary = formatMonetary;
+        useAutofocus({ refName: "searchInput", mobile: true });
+        this.productListDiv = useRef("productListDiv");
+        this.currentProductCard = useRef(`product_${this.state.currentProduct}`);
```

Comment:
If `state` is quite dynamic, then this might not work as intended because `setup` is only called once. Mutations in `state` won't necessarily mean recreation of the instance of this component.

Reply:
here `this.currentProductCard` should never change during the lifetime of the `ProductList` component. It signifies the  product whose page you last viewed, so it can only change if you go to the `ProductMainView` page.
Is it ok to leave it like this in this case?